e2e_deployment_files/connectivity_deployment.template.json

```diff
@@ -579,6 +597,36 @@
               "image": "<Container_Registry>/microsoft/azureiotedge-network-controller:<Build.BuildNumber>-linux-<Architecture>",
               "createOptions": "{\"HostConfig\":{\"Privileged\":\"true\",\"NetworkMode\":\"host\",\"Binds\":[\"/var/run/docker.sock:/var/run/docker.sock\"]},\"NetworkingConfig\":{\"EndpointsConfig\":{\"host\":{}}}}"
             }
+          },
+          "metricsCollector": {
+            "version": "1.0",
+            "type": "docker",
+            "status": "running",
+            "restartPolicy": "always",
+            "env": {
+                "logAnalyticsWorkspaceId": {
+                    "value": "<LogAnalyticsWorkspaceId>"
+                },
+                "logAnalyticsSharedKey": {
+                    "value": "<LogAnalyticsSharedKey>"
+                },
+                "logAnalyticsLogType": {
+                  "value": "testMetrics"
+                },
+                "MetricsEndpointsCSV": {
+                  "value": "<MetricsCollector.MetricsEndpointsCSV>"
+                },
+                "ScrapeFrequencyInSecs": {
+                    "value": "<MetricsCollector.ScrapeFrequencyInSecs>"
+                },
+                "UploadTarget": {
+                  "value": "<MetricsCollector.UploadTarget>"
+                }
+            },
+            "settings": {
+                "image": "<Container_Registry>/microsoft/azureiotedge-metrics-collector:<Build.BuildNumber>-linux-<Architecture>",
+                "createOptions": ""
```

Comment:
please use host network; and make sure to open port in edgeAgent and edgeHub of metric endpoints to the host.